lib/preview-frame.ts

```diff
@@ -5,12 +5,16 @@ require("../css/preview-frame.css");
 interface PreviewFrameWindow extends PreviewFrame.Runner {
   // This is exported by p5 when it's in global mode.
   noLoop: () => void;
+
+  // This is the p5 constructor. An undocumented feature is that
+  // even the first argument is actually *optional*; if omitted,
+  // p5 will initialize itself in global mode.
+  p5: (sketch?: Function, node?: HTMLElement, sync?: boolean) => void;
 }
 
 let global = window as PreviewFrameWindow;
 
-function loadP5(version: string, cb?: () => void) {
-  let url = '//cdnjs.cloudflare.com/ajax/libs/p5.js/' + version + '/p5.js';
+function loadScript(url, cb?: () => void) {
```

Comment:
Cool! Glad my drive-by review was helpful ^_^

On Monday, August 29, 2016, Atul Varma <notifications@github.com> wrote:

> In lib/preview-frame.ts
> <https://github.com/toolness/p5.js-widget/pull/63#discussion_r76695989>:
>
> >  }
> >
> >  let global = window as PreviewFrameWindow;
> >
> > -function loadP5(version: string, cb?: () => void) {
> > -  let url = '//cdnjs.cloudflare.com/ajax/libs/p5.js/' + version + '/p5.js';
> > +function loadScript(url, cb?: () => void) {
>
> Ah good idea! I'll add that--this will also be useful for when we allow
> people to specify arbitrary URLs to third-party add-on libraries, since we
> want to provide good feedback if they misspell any of those URLs or
> something.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/toolness/p5.js-widget/pull/63/files/ffe52963c21320e73664190dcfa70c2878b3a50e#r76695989>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAF4jGGEq79d0FNqZtg7XCRb6v1kKU5tks5qk1ZqgaJpZM4Ju51N>
> .
>